src/Agent/NewRelic/Agent/Core/Attributes/LogContextDataFilter.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NewRelic.Agent.Configuration;
+
+namespace NewRelic.Agent.Core.Attributes
+{
+    public interface ILogContextDataFilter
+    {
+        Dictionary<string, object> FilterLogContextData(Dictionary<string, object> contextData);
+    }
+
+    public class LogContextDataFilter : ILogContextDataFilter
+    {
+        private IConfigurationService _configurationService;
+        private List<LogContextDataFilterRule> _includeRuleList;
+        private List<LogContextDataFilterRule> _excludeRuleList;
+        private List<LogContextDataFilterRule> _orderedCludeRuleList;
+
+        private Dictionary<string, bool> _clusionCache = new Dictionary<string, bool>();
+
+        public LogContextDataFilter(IConfigurationService configurationService)
+        {
+            _configurationService = configurationService;
+        }
+
+
+        public Dictionary<string, object> FilterLogContextData(Dictionary<string, object> unfilteredContextData)
+        {
+            if (unfilteredContextData == null)
+            {
+                return null;
+            }
+
+            var filteredContextData = new Dictionary<string, object>();
+
+            // Pseudocode
+            // Include/exclude rules:
+            // 1. Exclude wins over include
+            // 2. Can have a '*' wildcard at the end of an include/exclude definition
+            // 3. "more specific" rules win over "less specific".  Example: include=AB, exclude=A*, AB would be included but not ABC or AAA
+
```

Comment:
Not serious comment: We could make this even more efficient if we used a TRIE! (don't refactor to use a TRIE)